llama-index-core/llama_index/core/indices/base.py

```diff
@@ -203,17 +203,23 @@ def insert_nodes(self, nodes: Sequence[BaseNode], **insert_kwargs: Any) -> None:
             self._insert(nodes, **insert_kwargs)
             self._storage_context.index_store.add_index_struct(self._index_struct)
 
-    def insert(self, document: Document, **insert_kwargs: Any) -> None:
+    def insert(
```

Comment:
Updating base class methods has some larger impacts -- need to check every class that subclasses `BaseIndex` and ensure that if its implementing this method, that it matches